hooks/read-package-hook/src/createVersionsOverrider.ts

```diff
@@ -103,10 +111,14 @@ function overrideDeps (
       )}`
       continue
     }
-    if (versionOverride.linkFileTarget) {
-      deps[
-        versionOverride.targetPkg.name
-      ] = `file:${versionOverride.linkFileTarget}`
+    if (versionOverride.linkFileTarget && dir) {
```

Comment:
What about when dir is undefined? I assume that would happen when the override is applied on a dependency. Probably and absolute path should be used in that case.

Reply:
https://github.com/pnpm/pnpm/pull/8055/commits/f1023f69fb335235a920c27f379b88408946b5c9